iwyu_preprocessor.cc

```diff
@@ -209,6 +217,22 @@ void IwyuPreprocessorInfo::HandlePragmaComment(SourceRange comment_range) {
     return;
   }
 
+  if (HasOpenBeginKeep(this_file_entry)) {
+    if (MatchOneToken(tokens, "end_keep", 1, begin_loc)) {
+      ERRSYM(this_file_entry) << "end_keep pragma seen\n";
+      SourceLocation keep_loc_begin = begin_keep_location_stack_.top();
+      begin_keep_location_stack_.pop();
+      keep_location_ranges_.insert({this_file_entry,
```

Comment:
Initialization syntax is rare in IWYU, I would prefer
```
SourceRange keep_range(keep_loc_begin, begin_loc);
keep_location_ranges_.insert(std::make_pair(this_file_entry, keep_range));
```
here, to avoid implicit typing and line wrapping.